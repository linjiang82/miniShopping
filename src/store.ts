import { createStore } from "redux";
import testReducer from "./reducers/reducer";

const store = createStore(testReducer);
export default store;
