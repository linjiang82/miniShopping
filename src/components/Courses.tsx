import React from 'react'
import Course from './Course'
import {connect} from 'react-redux'
import {TCourse, TRootReducer} from '../types/types'
import { db } from '../firebase/config'
import { fetchCourses } from '../actions/action'
import store from '../store';
import './Courses.css'

 interface myProps {
  courses:TCourse[];
}
class Courses extends React.Component<myProps>{
  componentDidMount(){
    db.collection('Courses').get().then((snapshot)=>snapshot.forEach(doc=>store.dispatch(fetchCourses({...doc.data(),id:doc.id})))).catch(e=>console.log("error"+e));
    // (async ()=> { const data = await db.collection('Courses').get(); 
    //             const items = await data.forEach(doc=>{console.log(doc.data())})})()
  }
  render(){
    let courses:JSX.Element[]=[];
    this.props.courses.map(course=>courses.push(<Course key={course.id} data={course}></Course>))
   if(this.props.courses.length>0) 
   console.log(this.props.courses.length) 
    return <div className='flexCtn'>
      { courses }
    </div>
  }
}

const mapStateToProps = (state: TRootReducer ):myProps => {
  return { courses: state.courses,}
  
}

export default connect(mapStateToProps)(Courses)