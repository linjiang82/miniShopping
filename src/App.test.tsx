// import React from 'react';
// import { render } from '@testing-library/react';
import Courses from "./components/Courses";
import React from "react";
import ReactDOM from "react-dom";
import store from "./store";
import fetchMock from "fetch-mock";
import {
  render,
  cleanup,
  fireEvent,
  waitForElement
} from "@testing-library/react";

afterEach(cleanup);
describe("<Course />", () => {
  test("renders 1 app", async () => {
    const props = {
      store: store
    };
    // const div = document.createElement("div");
    // ReactDOM.render(<Courses {...props} />, div);
    const { debug, asFragment } = render(<Courses {...props} />);
    expect(asFragment()).toMatchSnapshot();
    debug();
  });
});
const fetchdata = async () => {
  const res = await fetch("https://2");
  const final = await res.json();
  return final;
};
fetchMock.get("https://2", { id: 3 });
test("async with async", async () => {
  const data = await fetchdata();
  const id = data.id;
  expect(id).toBe(3);
});
test("async with promise", () => {
  return fetchdata().then(data => expect(data.id).toBe(3));
});
