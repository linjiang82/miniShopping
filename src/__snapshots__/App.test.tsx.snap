// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Course /> renders 1 app 1`] = `
<DocumentFragment>
  <div>
    <div />
  </div>
</DocumentFragment>
`;
